src/lib/utils/longRunningProcessTimer.ts

```diff
@@ -0,0 +1,121 @@
+import {PerformanceObserver, performance} from 'perf_hooks';
+import {red} from 'colors';
+const utils = require('./utils.js');
+import prettyMs from 'pretty-ms';
+import { Logger } from '../../../src/typings/logger';
+const ora = require('ora');
+
+export default class LongRunningProcessTimer{
+  private _logger: Logger;
+  private _packageName: string;
+  private _version: string;
+  private _showSpinner: boolean;
+  private _spinnerStyle: string;
+  private _interval: number;
+  private _processStartingMsg: string;
+  private _processOngoingMsg: string;
+  private _processFinishedMsg: string;
+  private _longRunningThreshold: number;
+  private _startMark: string;
+  private _ongoingMark: string;
+  private _downloadOngoing: string;
+  private _endMark: string;
+  private _downloadComplete: string;
+  private _observer!: PerformanceObserver;
+  private _spinner!: any;
+  private _intOngoingDownload!: NodeJS.Timeout;
+
+  constructor(logger: Logger, packageName: string, version: string, processStartingMsg: string, processOngoingMsg: string, processFinishedMsg: string, showSpinner?:boolean, spinnerStyle?: string, interval?: number, longRunningThreshold?: number){
+    this._logger = (logger && typeof logger.info === 'function') ? logger : console;
+    this._packageName = packageName;
+    this._version = version;
+    this._showSpinner = showSpinner || false;
+    this._spinnerStyle = spinnerStyle || "dots";
+    this._interval = interval || 750;
+    this._processStartingMsg = processStartingMsg;
+    this._processOngoingMsg = processOngoingMsg;
+    this._processFinishedMsg = processFinishedMsg;
+    this._longRunningThreshold = longRunningThreshold || 4000
+
+
+    // define mark and measurement names
+    this._startMark = 'downloadStart' + this._packageName + this._version;
+    this._ongoingMark = 'downloadOngoingMark' + this._packageName + this._version;
+    this._downloadOngoing = 'downloadOngoing' + this._packageName + this._version;
+    this._endMark = 'downloadEnd' + this._packageName + this._version;
+    this._downloadComplete = 'downloadComplete' + this._packageName + this._version;
+
+    this.observer.observe({entryTypes: ['measure']});
+  }
+
+  get observer(){
+    if(typeof this._observer === 'undefined'){
```

Comment:
Syntax sugar: I always prefer to use guard close when possible 
that avoid to have this:
```
if (something) {
  if (somethingElse) {
  }
}
```

So instead you can do:
```
if(typeof this._observer !== 'undefined'){
  return this._observer;
}
....
```

Reply:
Also, see my comment below in tslint.json

Reply:
It's fine it is just syntax sugar, we all have our preferences and habits. 👍